Templates/Integrations/DataEnrichmentThreatIntelligence/DataEnrichmentThreatIntelligence.py

```diff
@@ -0,0 +1,320 @@
+        """Performs basic get request to see if the API is reachable and authentication works.
+
+        Returns:
+            Response JSON
+        """
+        return self._http_request('GET', 'version')
+
+    def get_ip_request(self, ip: str) -> Dict:
+        """Gets an analysis from the API for given IP.
+
+        Args:
+            ip: IP to get analysis on
+
+        Returns:
+            Response JSON
+
+        """
+        suffix = 'ip'
+        params = {'ip': ip}
+        return self._http_request('GET', suffix, params=params)
+
+    def get_url_request(self, url: str) -> Dict:
+        """Gets an analysis from the API for given url.
+
+        Args:
+            url: URL to get analysis on
+
+        Returns:
+            Response JSON
+        """
+        suffix = 'analysis'
+        params = {'url': url}
+        return self._http_request('GET', suffix, params=params)
+
+    def search_file_request(self, file_hash: str) -> Dict:
+        """Building request for file command
+
+        Args:
+            file_hash: Hash to search in API
+
+        Returns:
+            Response JSON
+        """
+        suffix = 'analysis'
+        params = {'hash': file_hash}
+        return self._http_request('GET', suffix, params=params)
+
+    def get_domain_request(self, domain):
+        """Building request for file command
+
+        Args:
+            domain: Domain to search in API
+
+        Returns:
+            Response JSON
+        """
+        suffix = 'analysis'
+        params = {'domain': domain}
+        return self._http_request('GET', suffix, params=params)
+
+
+def build_context(results: Union[Dict, List], indicator_type: str) -> Union[Dict, List]:
+    """Formatting results from API to Demisto Context
+
+    Args:
+        results: raw results from raw_response
+        indicator_type: type of indicator
+
+    Returns:
+        Results formatted to Demisto Context
+    """
+
+    def build_entry_context(entry: Dict):
+        return {
+            'ID': entry.get('id'),
+            'Severity': entry.get('severity'),
+            indicator_type: entry.get('indicator'),
+            'Description': entry.get('description')
+        }
+
+    if isinstance(results, list):
+        return [build_entry_context(entry) for entry in results]  # pragma: no cover
+    return build_entry_context(results)
+
+
+''' COMMANDS '''
+
+
+def search_ip(client: Client, args: Dict) -> Tuple[str, Dict, Dict]:
+    """Gets results for the API.
+    """
+    ip = args.get('ip', '')
+    raw_response = client.get_ip_request(ip)
+    results = raw_response.get('result')
+    if results:
+        result = results[0]
+        title = f'{INTEGRATION_NAME} - Analysis results for IP: {ip}'
+        context_entry = build_context(result, 'IP')
+        # Building a score for DBot
+        score = client.calculate_dbot_score(result.get('severity'))
+        dbot_entry = build_dbot_entry(ip, 'ip', INTEGRATION_NAME, score, result.get('description'))
```

Comment:
whats `build_dbot_entry` ?

Reply:
it builds dbot entry according to given args.